state.go

```diff
@@ -83,20 +87,39 @@ func (r *raftState) setCurrentTerm(term uint64) {
 	atomic.StoreUint64(&r.currentTerm, term)
 }
 
-func (r *raftState) getLastLogIndex() uint64 {
-	return atomic.LoadUint64(&r.LastLogIndex)
+func (r *raftState) getLastLog() (index, term uint64) {
+	r.lastLock.Lock()
+	index = r.lastLogIndex
+	term = r.lastLogTerm
+	r.lastLock.Unlock()
+	return
+}
+
+func (r *raftState) getLastLogIndexOnly() uint64 {
```

Comment:
I'd prefer only the 2 item version. I think for future maintenance it's probably better to make people think about fetching them together as a unit as well.